src/decaylanguage/dec/dec.py

```diff
@@ -195,7 +196,11 @@ def parse(self, include_ccdecays: bool = True) -> None:
 
         # Instantiate the Lark parser according to chosen settings
         parser = Lark(
-            self.grammar(), parser=opts["parser"], lexer=opts["lexer"], **extraopts
+            self.grammar(),
+            parser=opts["parser"],
+            lexer=opts["lexer"],
+            edit_terminals=_edit_model_name_terminals,
```

Comment:
A disadvantage of hard-coding it like this is that `edit_terminals` cannot be supplied via `extraopts` anymore. If you see a need to still be able to do that I'm sure I could try to come up with something that combines all callbacks somehow.

Reply:
Hey. You mean if somebody was to fully define the parser? I added all that flexibility at the beginning but might remove it at some point. The fact is that I believe nobody has ever used it and there are no tests to ensure other Lark parsers do the job right, and wisdom says that if something is not tested, it is not in practice supported.

Is this what you meant?

Reply:
Yes, exactly. I mean if someone is currently passing `edit_terminals` to `load_grammar()` that would end up in `extraopts` and clash with this additional argument. I'm not sure anyone really does this but since `load_grammar()` is a public method I thought I'd check with you.

Reply:
Hmm. If I remember correctly, the main/real reason why I added `load_grammar` was to be able to easily define an updated grammar, typically for a new model, and then continue transparently. Now, if the models aren't in the grammar anymore, than makes far less sense.

It would be useful to see a couple of use-cases spelled out, see my comment https://github.com/scikit-hep/decaylanguage/pull/390#issuecomment-1804349189 below. The way you suggest, which is sounding good to me, we should then deprecate `load_grammar` with a reference to this PR so that people can understand the rationale. I don't think we need to give much advance warning - basically the next minor version 0.18.0 - since, as said already, I do not think anyone is in fact using this feature. Besides, we now are in a much much better situation of supporting all LHCb and Belle II models we are aware of as of today. It is unlikely that anyone using a fresh new model won't be able to wait 1-2 days max for a new release with the new model tested ;-).

Reply:
Hi. Let's try and get back to this.

Again, I understand why you are suggesting this PR. The removal of duplication sounds fair. Going in this direction, you need to remove the `load_grammar` method. This package is not used by tons of people, hence acceptable, I reckon.

Now, if we wanted to still give the flexibility for a user to provide a list of models, that would need to be done via a public method. What do you think?